cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+	}
+	currenttime := time.Now().Unix()
+
+	// If we're transitioning state, update the metadata
+	if isTrusted(newnode) != newstate {
+		if newstate == false {
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/untrustedsince"] = strconv.FormatInt(currenttime, 10)
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/trustedsince"] = ""
+		} else {
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/untrustedsince"] = ""
+			newnode.ObjectMeta.Annotations["tpm.coreos.com/trustedsince"] = strconv.FormatInt(currenttime, 10)
+		}
+	}
+
+	newnode.ObjectMeta.Annotations["tpm.coreos.com/validationtime"] = strconv.FormatInt(currenttime, 10)
+
+	// Ensure that the new node is tainted appropriately
+	if newstate == true {
+		validateNode(newnode)
+	} else {
+		invalidateNode(newnode)
+	}
+	newnode.ObjectMeta.Annotations["tpm.coreos.com/logstate"] = node.ObjectMeta.Annotations["tpm.coreos.com/logstate"]
+	newnode, err = manager.client.Nodes().Update(newnode)
+	if err != nil {
+		glog.Errorf("Unable to update node state for %s: %v", node.Name, err)
+		return
+	}
+}
+
+func verifyAllNodes() {
+	nodes, err := manager.client.Nodes().List(api.ListOptions{})
+	if err != nil {
+		return
+	}
+	for _, node := range nodes.Items {
+		verifyAndUpdate(&node)
+	}
+}
+
+func reverify() {
+	select {
+	case <-time.After(time.Duration(manager.recurring) * time.Second):
+		verifyAllNodes()
+	case <-manager.recurringChan:
+	}
+}
+
+type TPMManager struct {
+	Master        string
+	Kubeconfig    string
+	tpmhandler    tpm.TPMHandler
+	pcrconfig     string
+	pcrconfigdir  string
+	allowunknown  bool
+	recurring     int
+	client        *client.Client
+	policyTimer   *time.Timer
+	leaderelect   componentconfig.LeaderElectionConfiguration
+	recurringChan chan int
+	pcrconfigs    []map[string]tpm.PCRConfig
+}
+
+var manager TPMManager
+
+func addFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&manager.Master, "master", manager.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig)")
+	fs.StringVar(&manager.Kubeconfig, "kubeconfig", manager.Kubeconfig, "Path to kubeconfig file with authorization and master location information.")
+	fs.StringVar(&manager.pcrconfig, "pcrconfig", manager.pcrconfig, "Path to a single PCR config file")
+	fs.StringVar(&manager.pcrconfigdir, "pcrconfigdir", manager.pcrconfigdir, "Path to a PCR config directory")
+	fs.BoolVar(&manager.allowunknown, "allowunknown", false, "Allow unknown TPMs to join the cluster")
+	fs.IntVar(&manager.recurring, "reverify", 0, "Periodocally reverify nodes after this many seconds")
+}
+
+func updateConfig(configmap *api.ConfigMap) {
+	if configmap.Data["allowunknown"] != "" {
+		allowunknown, err := strconv.ParseBool(configmap.Data["allowunknown"])
+		if err == nil {
+			manager.allowunknown = allowunknown
+		}
+	}
+	if configmap.Data["reverify"] != "" {
+		reverify, err := strconv.Atoi(configmap.Data["reverify"])
+		if err == nil {
+			manager.recurring = reverify
+			select {
```

Comment:
Rather then make it non-blocking, won't this actually cause the channel send to not complete if the channel is blocked? Not sure if this is intended.

If you want just non-blocking I think you might want
```
go func() {
    channel <- 1
}
```

I'm not familiar with what is on the other end of the channel either or how often it is drained. Maybe just a buffered channel or allowing it to block is ok too.